java/runtime/src/main/java/org/ray/runtime/gcs/GcsClientImpl.java

```diff
@@ -0,0 +1,130 @@
+package org.ray.runtime.gcs;
+
+import com.google.common.base.Preconditions;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.ArrayUtils;
+import org.ray.api.Checkpointable.Checkpoint;
+import org.ray.api.gcs.GcsClient;
+import org.ray.api.gcs.NodeInfo;
+import org.ray.api.id.UniqueId;
+import org.ray.runtime.generated.ActorCheckpointIdData;
+import org.ray.runtime.generated.ClientTableData;
+import org.ray.runtime.generated.TablePrefix;
+import org.ray.runtime.util.UniqueIdUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An implementation of GcsClient.
+ */
+public class GcsClientImpl implements GcsClient {
+
+  private static Logger LOGGER = LoggerFactory.getLogger(GcsClientImpl.class);
+
+  private RedisClient primary;
+
+  private List<RedisClient> shards;
+
+  public GcsClientImpl(String redisAddress, String redisPassword) {
+    primary = new RedisClient(redisAddress, redisPassword);
+    int numShards = 0;
+    try {
+      numShards = Integer.valueOf(primary.get("NumRedisShards", null));
+      Preconditions.checkState(numShards > 0,
+          String.format("Expected at least one Redis shards, found %d.", numShards));
+    } catch (NumberFormatException e) {
+      throw new RuntimeException("Failed to get number of redis shards.", e);
+    }
+
+    List<byte[]> shardAddresses = primary.lrange("RedisShards".getBytes(), 0, -1);
+    Preconditions.checkState(shardAddresses.size() == numShards);
+    shards = shardAddresses.stream().map((byte[] address) -> {
+      return new RedisClient(new String(address));
+    }).collect(Collectors.toList());
+  }
+
+  /**
+   * If the actor exists in GCS.
+   */
+  public boolean actorExists(UniqueId actorId) {
```

Comment:
Let's put this method after `getAllNodeInfo`, so all the public APIs (in the `GcsClient` interface) come first before the private APIs